test/python/algorithms/minimum_eigensolvers/test_qaoa.py

```diff
@@ -0,0 +1,353 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test the QAOA algorithm."""
+
+import unittest
+from test.python.algorithms import QiskitAlgorithmsTestCase
+
+# from functools import partial
+import numpy as np
+
+# from scipy.optimize import minimize as scipy_minimize
+from ddt import ddt, idata, unpack
+
+# import retworkx as rx
+
+from qiskit import QuantumCircuit  # , QuantumRegister
+from qiskit.algorithms.minimum_eigensolvers import QAOA
+from qiskit.algorithms.optimizers import COBYLA  # , NELDER_MEAD
+from qiskit.circuit import Parameter
+from qiskit.opflow import PauliSumOp, I, X  # , Z
```

Comment:
I would remove as many opflow dependencies as possible, you can later define these as `PauliSumOp`s
```suggestion
from qiskit.opflow import PauliSumOp
```